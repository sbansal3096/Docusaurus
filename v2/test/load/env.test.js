import path from 'path';
import loadEnv from '@lib/load/env';

describe('loadEnv', () => {
  test('website with both versioning & translation disabled', () => {
    const siteDir = path.join(__dirname, '__fixtures__', 'simple-site');
    const siteConfig = {
      baseUrl: '/',
      organizationName: 'endiliey',
      projectName: 'hello',
      tagline: 'Hello World',
      title: 'Hello',
    };
    const env = loadEnv({siteDir, siteConfig});
    expect(env.versioning.enabled).toBe(false);
    expect(env.translation.enabled).toBe(false);
    expect(env).toMatchSnapshot();
  });

  test('website with versioning enabled', () => {
    const siteDir = path.join(__dirname, '__fixtures__', 'versioned-site');
    const siteConfig = {
      baseUrl: '/',
      organizationName: 'endiliey',
      projectName: 'hello',
      tagline: 'Hello World',
      title: 'Hello',
    };
    const env = loadEnv({siteDir, siteConfig});
    expect(env.versioning.enabled).toBe(true);
    expect(env.translation.enabled).toBe(false);
    expect(env).toMatchSnapshot();
  });

  test('website with translation enabled', () => {
    const siteDir = path.join(__dirname, '__fixtures__', 'translated-site');
    const siteConfig = {
      baseUrl: '/',
      organizationName: 'endiliey',
      projectName: 'hello',
      tagline: 'Hello World',
      title: 'Hello',
      defaultLanguage: 'en',
    };
    const env = loadEnv({siteDir, siteConfig});
    expect(env.versioning.enabled).toBe(false);
    expect(env.translation.enabled).toBe(true);
    expect(env).toMatchSnapshot();
  });

  test('website with versioning & translation enabled', () => {
    const siteDir = path.join(__dirname, '__fixtures__', 'transversioned-site');
    const siteConfig = {
      baseUrl: '/',
      organizationName: 'endiliey',
      projectName: 'hello',
      tagline: 'Hello World',
      title: 'Hello',
      defaultLanguage: 'en',
    };
    const env = loadEnv({siteDir, siteConfig});
    expect(env.versioning.enabled).toBe(true);
    expect(env.translation.enabled).toBe(true);
    expect(env).toMatchSnapshot();
  });

  test('website with languages.js but no lang is enabled', () => {
    const siteDir = path.join(__dirname, '__fixtures__', 'bad-site');
    const siteConfig = {
      baseUrl: '/',
      organizationName: 'endiliey',
      projectName: 'hello',
      tagline: 'Hello World',
      title: 'Hello',
    };
    expect(() => {
      loadEnv({siteDir, siteConfig});
    }).toThrowErrorMatchingInlineSnapshot(
      `"Please at least enable one language in 'languages.js'"`,
    );
  });

  test('website with languages.js but no default language set', () => {
    const siteDir = path.join(__dirname, '__fixtures__', 'translated-site');
    const siteConfig = {
      baseUrl: '/',
      organizationName: 'endiliey',
      projectName: 'hello',
      tagline: 'Hello World',
      title: 'Hello',
    };
    expect(() => {
      loadEnv({siteDir, siteConfig});
    }).toThrowErrorMatchingInlineSnapshot(
      `"Please set a default language in 'siteConfig.js' which is enabled in 'languages.js'"`,
    );
  });
});
