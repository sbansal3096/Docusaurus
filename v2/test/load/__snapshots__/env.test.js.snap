// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadEnv website with both versioning & translation disabled 1`] = `
Object {
  "translation": Object {
    "defaultLanguage": Object {},
    "enabled": false,
    "enabledLanguages": Array [],
  },
  "versioning": Object {
    "defaultVersion": null,
    "enabled": false,
    "latestVersion": null,
    "versions": Array [],
  },
}
`;

exports[`loadEnv website with translation enabled 1`] = `
Object {
  "translation": Object {
    "defaultLanguage": Object {
      "enabled": true,
      "name": "English",
      "tag": "en",
    },
    "enabled": true,
    "enabledLanguages": Array [
      Object {
        "enabled": true,
        "name": "English",
        "tag": "en",
      },
      Object {
        "enabled": true,
        "name": "Español",
        "tag": "es-ES",
      },
      Object {
        "enabled": true,
        "name": "Română",
        "tag": "ro",
      },
      Object {
        "enabled": true,
        "name": "Türkçe",
        "tag": "tr",
      },
      Object {
        "enabled": true,
        "name": "简体中文",
        "tag": "zh-CN",
      },
    ],
  },
  "versioning": Object {
    "defaultVersion": null,
    "enabled": false,
    "latestVersion": null,
    "versions": Array [],
  },
}
`;

exports[`loadEnv website with versioning & translation enabled 1`] = `
Object {
  "translation": Object {
    "defaultLanguage": Object {
      "enabled": true,
      "name": "English",
      "tag": "en",
    },
    "enabled": true,
    "enabledLanguages": Array [
      Object {
        "enabled": true,
        "name": "English",
        "tag": "en",
      },
      Object {
        "enabled": true,
        "name": "Español",
        "tag": "es-ES",
      },
      Object {
        "enabled": true,
        "name": "Română",
        "tag": "ro",
      },
      Object {
        "enabled": true,
        "name": "Türkçe",
        "tag": "tr",
      },
      Object {
        "enabled": true,
        "name": "简体中文",
        "tag": "zh-CN",
      },
    ],
  },
  "versioning": Object {
    "defaultVersion": "1.0.1",
    "enabled": true,
    "latestVersion": "1.0.1",
    "versions": Array [
      "1.0.1",
      "1.0.0",
    ],
  },
}
`;

exports[`loadEnv website with versioning enabled 1`] = `
Object {
  "translation": Object {
    "defaultLanguage": Object {},
    "enabled": false,
    "enabledLanguages": Array [],
  },
  "versioning": Object {
    "defaultVersion": "1.0.1",
    "enabled": true,
    "latestVersion": "1.0.1",
    "versions": Array [
      "1.0.1",
      "1.0.0",
    ],
  },
}
`;
